src/iface/mod.rs

```diff
@@ -23,4 +28,9 @@ pub use self::interface::{
 };
 
 pub use self::route::{Route, RouteTableFull, Routes};
+#[cfg(all(
+    feature = "proto-ipv6",
+    any(feature = "medium-ethernet", feature = "medium-ieee802154")
```

Comment:
SLAAC is a general IPv6 thing, why restrict it to any particular medium(s)?

Reply:
SLAAC generates an IP addr based on the MAC addr. With IP medium there's no MAC addr. How is it supposed to work?

Reply:
as @Dirbaio mentioned, there needs to be support for each link layer with SLAAC, see for example [rfc 2464](https://www.rfc-editor.org/rfc/rfc2464.html#section-4) and [rfc 4944](https://www.rfc-editor.org/rfc/rfc4944#section-6).

Reply:
Sure for link-local address you want to use the MAC address (or equivalent) to generate it, but that's not quite the same for global or temporary addresses where you might want to hide your MAC address. I think in this case the user should be allowed to specify an interface identifier (which would be different from the MAC).

If Linux can do SLAAC for TUN devices, smoltcp can definitely do it for IP mediums.

Reply:
> Or does it always add a privacy address to that interface for slaac?

Yes

> I prefer to leave the additional logic required to switch the generation of interface identifiers to a follow up PR as this PR is already complex enough.

Sure the additional logic isn't necessary for this PR, I'm just saying the `any(feature = "medium-ethernet", feature = "medium-ieee802154")` isn't needed.